packages/terra-form-radio/src/Radio.jsx

```diff
@@ -0,0 +1,178 @@
+
+const propTypes = {
+  /**
+    * Whether or not the radio button is checked. Use this to generate a controlled radio button.
+    */
+  checked: PropTypes.bool,
+  /**
+    * The checked property of the Input element. Use this to generate an uncontrolled radio button.
+    */
+  defaultChecked: PropTypes.bool,
+  /**
+    * The id of the radio button.
+    */
+  id: PropTypes.string,
+  /**
+    * Additional attributes for the input object.
+    */
+  inputAttrs: PropTypes.object,
+  /**
+    * Whether the radio button is disabled.
+    */
+  disabled: PropTypes.bool,
+  /**
+    * Whether the radio button is inline.
+    */
+  isInline: PropTypes.bool,
+  /**
+    * Whether the label is hidden.
+    */
+  isLabelHidden: PropTypes.bool,
+  /**
+    * Text of the label.
+    */
+  labelText: PropTypes.node.isRequired,
+  /**
+    * Additional attributes for the text object.
+    */
+  labelTextAttrs: PropTypes.object,
+  /**
+    * Name attribute of the input.
+    */
+  name: PropTypes.string,
+  /**
+   * Function to trigger when focus is lost from the radio button.
+   */
+  onBlur: PropTypes.func,
+  /**
+    * Function to trigger when user clicks on the radio button. Provide a function to create a controlled input.
+    */
+  onChange: PropTypes.func,
+    /**
+   *  Function to trigger when user focuses on the radio button.
+   */
+  onFocus: PropTypes.func,
+  /**
+    * The value of the input element.
+    */
+  value: PropTypes.string,
+};
+
+const defaultProps = {
+  checked: undefined,
+  defaultChecked: undefined,
+  id: undefined,
+  inputAttrs: {},
+  disabled: false,
+  isInline: false,
+  isLabelHidden: false,
+  labelText: null,
+  labelTextAttrs: {},
+  name: null,
+  onBlur: undefined,
+  onChange: undefined,
+  onFocus: undefined,
+  value: undefined,
+};
+
+const Radio = ({
+    checked,
+    defaultChecked,
+    inputAttrs,
+    id,
+    disabled,
+    isInline,
+    isLabelHidden,
+    labelText,
+    labelTextAttrs,
+    name,
+    onBlur,
+    onChange,
+    onFocus,
+    value,
+    ...customProps
+  }) => {
+  const controlInputAttrs = Object.assign({}, inputAttrs);
+
+  if (checked !== undefined) {
+    controlInputAttrs.checked = checked;
+  } else {
+    controlInputAttrs.defaultChecked = defaultChecked;
+  }
+
+  const radioClasses = cx([
+    'radio',
+    { 'is-inline': isInline },
+    customProps.className,
+  ]);
+
+  const labelClasses = cx([
+    'label',
+    { 'is-disabled': disabled },
+    { 'is-hidden': isLabelHidden },
+    { 'is-mobile': RadioUtil.isConsideredMobileDevice() },
+    labelTextAttrs.className,
+  ]);
+
+  const inputClasses = cx([
+    'native-input',
+    inputAttrs.className,
+  ]);
+
+  const labelTextClasses = cx([
+    'label-text',
+  ]);
+
+  const outerRingClasses = cx([
+    'outer-ring',
+    { 'is-mobile': RadioUtil.isConsideredMobileDevice() },
+  ]);
+
+  const innerRingClasses = cx([
+    'inner-ring',
+  ]);
+
+  let labelTextContainer = null;
+  if (isLabelHidden) {
```

Comment:
Could we add this check in the classNames and apply a style of `display:none` instead of toggling if we apply the content?

Reply:
Currently, the `.label.is-hidden` css class just makes sure the outer `label` is large enough to contain the radio. Both ways of hiding the label textwork: making the `span` that contains the label text display to none, or not rendering the labeltext in the first place. I know we had some discussion in Checkbox about this. Anyone else have thoughts?